pkg/controllers/vdb/podfacts.go

```diff
@@ -366,6 +369,14 @@ func (p *PodFacts) collectPodByStsIndex(ctx context.Context, vdb *vapi.VerticaDB
 			return err
 		}
 		pf.hasNMASidecar = vk8s.HasNMAContainer(&pod.Spec)
+		// we get the sandbox name from the sts labels if the subcluster
+		// belongs to a sanbox. If the node is up, we will later retrieve
+		// the sandbox state from the catalog
+		val, ok := sts.Labels[vmeta.SandboxNameLabel]
+		if ok {
+			pf.sandbox = val
+			pf.isPrimary = true
```

Comment:
This is temporary. We cannot rely on the subcluster object to get the type once the node belongs to a sandbox. How are we going to get the subcluster type of a sandboxed node when it is down? Using the sandboxes statuses in vdb? Adding a label to the configmap? From the doc, the 1st sc to be added to the sandbox is the primary and the other will be secondary so all we need is that primary sc and we can deduce the type of other sc. Unless there is something I am missing, still trying to understand sandbox.

Reply:
Lets defer how we are going to figure this out. I think it's a reasonable assumption to make to start. Can we add a check somewhere that says you can only have a single subcluster in the sandbox? I would suggest we move this to a function with lots of comments too so that its clear it is a work in progress.